public-api/src/main/java/org/pmiops/workbench/publicapi/GenomicsController.java

```diff
@@ -38,6 +50,75 @@ public GenomicsController(AchillesAnalysisService achillesAnalysisService, CdrVe
         return ResponseEntity.ok(achillesAnalysisService.findAnalysisByIdAndDomain(3000L, "Genomics"));
     }
 
+    @Override
+    public ResponseEntity<Long> getVariantSearchResultSize(String variantSearchTerm) {
+        try {
+            cdrVersionService.setDefaultCdrVersion();
+        } catch(NullPointerException ie) {
+            throw new ServerErrorException("Cannot set default cdr version");
+        }
+        String COUNT_SQL_TEMPLATE = "";
+        String genomicRegionRegex = "(?i)(chr([0-9]{1,})*[XY]*:).*";
+        String variantIdRegex = "(?i)(\\d{1,}|X|Y)-\\d{5,}-[A,C,T,G]{1,}-[A,C,T,G]{1,}";
+
+        // Make sure the search term is not empty
+        if (!Strings.isNullOrEmpty(variantSearchTerm)) {
+            // Check if the search term matches genomic region search term pattern
+            if (variantSearchTerm.matches(genomicRegionRegex)) {
+                String[] regionTermSplit = variantSearchTerm.split(":");
+                COUNT_SQL_TEMPLATE = "SELECT count(*) as count\n" + "FROM `${projectId}.${dataSetId}.wgs_variant_mv` where contig= \"" + regionTermSplit[0] + "\"";
+                if (regionTermSplit.length > 1) {
+                    String[] rangeSplit = regionTermSplit[1].split("-");
+                    try {
+                        if (rangeSplit.length == 2) {
+                            Long low = Math.min(Long.valueOf(rangeSplit[0]), Long.valueOf(rangeSplit[1]));
+                            Long high = Math.max(Long.valueOf(rangeSplit[0]), Long.valueOf(rangeSplit[1]));
+                            COUNT_SQL_TEMPLATE = "SELECT count(*) as count\n" +
+                                    "FROM `${projectId}.${dataSetId}.wgs_variant_mv` where contig= \"" + regionTermSplit[0] + "\" and position <= " + high + " and position >= " + low + "\n";
+                        }
+                    } catch(NumberFormatException e) {
+                        System.out.println("Trying to convert bad number.");
+                    }
+                }
+            } else if (variantSearchTerm.matches(variantIdRegex)) {
+                // Check if the search term matches variant id pattern
+                COUNT_SQL_TEMPLATE = "SELECT count(*) as count\n" + "FROM `${projectId}.${dataSetId}.wgs_variant_mv` where variant_id= \"" + variantSearchTerm + "\"";
```

Comment:
private static final String WHERE_VARIANT_ID = " where variant_id = @variantId";
Then you sql statement can be something like:
String finalSql = COUNT_SQL_TEMPLATE + WHERE_VARIANT_ID;